sdk/src/main/java/com/sygic/travel/sdk/auth/service/AuthStorageService.kt

```diff
@@ -0,0 +1,74 @@
+package com.sygic.travel.sdk.auth.service
+
+import android.content.SharedPreferences
+import java.util.Date
+import java.util.UUID
+
+
+class AuthStorageService(private val sharedPreferences: SharedPreferences) {
+
+	private val CLIENT_SESSION = "clientSession"
+	private val USER_SESSION = "userSession"
+	private val DEVICE_ID = "deviceId"
+	private val EXPIRES_IN = "clientSession"
+	private val REFRESH_TOKEN_KEY = "refreshToken"
+
+	fun setUserSession(accessToken: String?) {
+		val editor = sharedPreferences.edit()
+		editor.putString(USER_SESSION, accessToken)
+		editor.apply()
+	}
+
+	fun getUserSession(): String? {
+		return sharedPreferences.getString(USER_SESSION, null)
+	}
+
+	fun setClientSession(accessToken: String?) {
+		val editor = sharedPreferences.edit()
+		editor.putString(CLIENT_SESSION, accessToken)
+		editor.apply()
+	}
+
+	fun getClientSession(): String? {
+		return sharedPreferences.getString(CLIENT_SESSION, null)
+	}
+
+	fun getDeviceId(): String? {
+		val editor = sharedPreferences.edit()
+		var deviceId = sharedPreferences.getString(DEVICE_ID, null)
+		if (deviceId.isNullOrBlank()) {
+			val uuid = UUID.randomUUID().toString()
+			editor.putString(DEVICE_ID, uuid)
+			editor.apply()
+			deviceId = uuid
+		}
+		return deviceId
+	}
+
+	fun setTokenRefreshTime(secondsToExpiration: Long) {
```

Comment:
I would store the real data, only add a getter called somehow... getTokenOptimalRefreshTime()

Reply:
Real data means you separately stores `currentDateTime` and `secondsToExpiration` a then in getter you retrieve both, do the math and return result ?

Reply:
No, I mean I would store the the real expiration timestamp, not the timestamp with 1/4 of the expiration time.

Reply:
Ok, and when I do store real expiration, how do I calculate when the token needs to be refreshed ? I need to know the length of the expiration, it needs to be stored (+1 stored item in SharedPreferences). Then in getter I subtract `realExpiration - (3/4* secondsToExpiration)`. I don't know, I can do it that way if you insist on it, but for me it storing the real expiration timestamp is useless because I don't need it.

Reply:
I see, I didn't undestand what you have meant by the previous comment.  Ok, what about storing the both of theam: real timestamp and suggested invalidation timestamp?

Reply:
:heavy_check_mark: